remindme-api/src/main/java/it/andreafailli/remindme/api/auth/FirebaseAuthenticationToken.java

```diff
@@ -0,0 +1,52 @@
+package it.andreafailli.remindme.api.auth;
+
+import java.util.Collection;
+
+import org.springframework.security.authentication.AbstractAuthenticationToken;
+import org.springframework.security.core.GrantedAuthority;
+
+import com.google.firebase.auth.FirebaseToken;
+
+public class FirebaseAuthenticationToken extends AbstractAuthenticationToken {
+
+	private static final long serialVersionUID = -1869548132546750002L;
+	
+	private final Object principal;
+	
+	private FirebaseToken credentials;
+
+	public FirebaseAuthenticationToken(Object principal, FirebaseToken credentials) {
+		super(null);
+		this.principal = principal;
+		this.credentials = credentials;
+		super.setAuthenticated(false);
+	}
+
+	public FirebaseAuthenticationToken(Object principal, FirebaseToken credentials, Collection<? extends GrantedAuthority> authorities) {
+		super(authorities);
+		this.principal = principal;
+		this.credentials = credentials;
+		super.setAuthenticated(true);
+	}
+	
+	public FirebaseToken getCredentials() {
+		return this.credentials;
+	}
+
+	public Object getPrincipal() {
+		return this.principal;
+	}
+
+	public void setAuthenticated(boolean isAuthenticated) throws IllegalArgumentException {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Add the "@Override" annotation above this method signature [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1161)
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove the declaration of thrown exception 'java.lang.IllegalArgumentException' which is a runtime exception. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)